lib/Config.php

```diff
@@ -435,4 +435,28 @@ public static function getConfigInstance(string $file): array
 
         throw new Exception($file . ' is invalid');
     }
+
+    public static function getDownloadFilenameByConfig(string $filename, string $type, string $default): string
+    {
+        $file = explode('.', $filename);
+
+        $naming = Config::getSystemConfiguration('download_naming')[$type];
+        if ($naming === 'default') {
+            $naming = $default;
+        } elseif (str_contains($filename, '%')) {
+            $filename = urldecode($filename);
+        }
+
+        $downloadFilename = str_replace(end($file), '', $filename);
```

Comment:
I think this can be make mistakes if the file extension is also in the filename. Why did you not use [pathinfo()](https://www.php.net/manual/de/function.pathinfo.php)?